.github/workflows/lint.yaml

```diff
@@ -0,0 +1,25 @@
+name: Lint
+on:
+    pull_request:
+        branches: [develop]
+jobs:
+    build:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v2
+            - uses: c-hive/gha-yarn-cache@v2
+            - name: Install Deps
+              run: "./scripts/ci/install-deps.sh --ignore-scripts"
+            - name: Typecheck
+              run: "yarn run lint:types"
+            - name: Switch js-sdk to release mode
+              run: |
+                  scripts/ci/js-sdk-to-release.js
+                  pushd node_modules/matrix-js-sdk
+                  yarn install
+                  yarn run build:compile
+                  yarn run build:types
+                  popd
+            - name: Typecheck (release mode)
```

Comment:
The `popd` is useless as each step runs in their own shell instances. The `pushd` can be replaced with `cd` which is enough (and easier to understand for a newcomer). If you didn’t have the script in the same step, you could even use `working-directory` (see [the documentation](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions?query=steps#jobsjob_idstepsrun)).

Reply:
oh great, even better :)